### PR TITLE
MC-38025: FPT disappears when switching back to the cart page

### LINKED ISSUE #30250: FPT disappears when switching back to the cart page
### Steps to reproduce:
 - Create 2 websites <ins> stores </ins> store views
 - Set price scope to website catalog > catalog > price > catalog price stope
 - Configure Fixed Product Taxes as below. stores > config > sales > tax > Fixed Product Taxes
image

 - Create an FPT attribute and assign it to the default attibute set
 <https://docs.magento.com/user-guide/tax/fixed-product-tax-configuration.html#step-2-create-an-fpt-attribute>
 - Create 4 simple products and set the FPT attribute value to Australia
 - Create 2 bundled products with following config

 - Define FPT
 - Dynamic price = no
 - set price to 100
 - bundle options shipped together, all marked as default with price type = fixed
 - add 2 of the simple products created in previous step.
 - Create a user account in the front end. Update the address with Australian address (set country to Australia or what ever the country used in FPT setup)
 - Add the 2 bundled products to the cart
 - Go to cart page and check the FPT is displaying correctly.
image

 - Click checkout with multiple address or proceed to checkout
 - click on back and look at cart page FPT disappears

### Actual result:

– FPT disappears
### (/) Expected result:

– FPT should be present

### PR DESCRIPTION
**Pull request by engcom-Golf**
### Description (*)

Used quote items in weee tax total calculation instead of quote address items

### Fixed Issues (if relevant)

1. Fixes magento/magento2#30250

### Manual testing scenarios (*)

See magento/magento2#30250
Expected result:
FPT tax value shouldn't disappear from totals block

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by engcom-Golf** [comment]
@magento run all tests

**Comment by engcom-Golf** [comment]
@magento run all tests

File: app/code/Magento/Weee/Model/Total/Quote/WeeeTax.php [hunk 1]
```diff
@@ -226,14 +233,13 @@ protected function processTotalAmount(
...
+     * @param Quote $quote
+     * @param Total $total
      * @return array
      */
-    public function fetch(\Magento\Quote\Model\Quote $quote, \Magento\Quote\Model\Quote\Address\Total $total)
+    public function fetch(Quote $quote, Total $total)
     {
-        /** @var $items \Magento\Sales\Model\Order\Item[] */
-        $items = isset($total['address_quote_items']) ? $total['address_quote_items'] : [];
+        $items = isset($total['quote_items']) ? $total['quote_items'] : [];
```
Comment (engcom-Kilo) [inline app/code/Magento/Weee/Model/Total/Quote/WeeeTax.php]: Please simplify this to $items = $total['quote_items'] ?? [];

File: dev/tests/integration/testsuite/Magento/Weee/Model/Total/Quote/CalculateTest.php [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    public function testGetWeeTaxTotals()
+    {
+        /** @var QuoteFactory $quoteFactory */
+        $quoteFactory = $this->objectManager->get(QuoteFactory::class);
+        /** @var QuoteResource $quoteResource */
+        $quoteResource = $this->objectManager->get(QuoteResource::class);
+        $quote = $quoteFactory->create();
+        $quoteResource->load($quote, 'multishipping_fpt_quote_id', 'reserved_order_id');
+        $cartId = $quote->getId();
+        $addressFactory = $this->objectManager->get(\Magento\Quote\Model\Quote\AddressFactory::class);
```
Comment (engcom-Kilo) [inline dev/tests/integration/testsuite/Magento/Weee/Model/Total/Quote/CalculateTest.php]: Please simplify FQN.

**Review by engcom-Kilo — APPROVED** [review]

**Comment by engcom-Golf** [comment]
@magento run all tests

**Comment by engcom-Bravo** [comment]
:heavy_check_mark: QA Passed

### Manual testing scenario
1. Create 2 websites, stores, store views
2. Set price scope to website catalog > catalog > price > catalog price stope
3. Configure Fixed Product Taxes as below. stores > config > sales > tax > Fixed Product Taxes
image
4. Create an FPT attribute and assign it to the default attribute set
 <https://docs.magento.com/user-guide/tax/fixed-product-tax-configuration.html#step-2-create-an-fpt-attribute>
5. Create 4 simple products and set the FPT attribute value to Australia
6. Create 2 bundled products with the following config
 - Define FPT
 - Dynamic price = no
 - set price to 100
 - bundle options shipped together, all marked as default with price type = fixed
 - add 2 of the simple products created in the previous step.
7. Create a user account in the front end. Update the address with Australian address (set country to Australia, country which is used in FPT setup)
8. Add the 2 bundled products to the cart
9. Go to cart page and check the FPT is displaying correctly.
10. Click checkout with multiple addresses or proceed to checkout
11. Click on back and look at cart page FPT disappears

**BEFORE** applying changes provided in the PR, the FPT disappeared
before fix

**AFTER** switching to the PR, the FPT is included into the Summary
After fix

The order is placed successfully and the FPT is charged

**Review by coderimus — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @coderimus, thank you for the review.
[ENGCOM-8612](https://jira.corp.magento.com/browse/ENGCOM-8612) has been created to process this Pull Request
:eight_spoked_asterisk: @coderimus, could you please add one of the following labels to the Pull Request?

| Label | Description |
|--------------------------------|-----------------------------------------------------------------|
| ```Auto-Tests: Covered``` | All changes in Pull Request is covered by auto-tests |
| ```Auto-Tests: Not Covered``` | Changes in Pull Request requires coverage by auto-tests |
| ```Auto-Tests: Not Required``` | Changes in Pull Request does not require coverage by auto-tests |

**Comment by zakdma** [comment]
@magento import code to https://github.com/magento-tsg/magento2ce

**Comment by magento-engcom-team** [comment]
@zakdma the branch with code successfully imported into`magento-tsg/magento2ce` repository. Branch name: `imported-magento-magento2-31563`.

**Pull request merged**